src/ansys/fluent/core/quantity.py

```diff
@@ -882,11 +883,18 @@ def __init__(self, real_value, unit_str=None, quantity_map=None):
         self._si_unit = self._unit.si_unit
         self._type = self._unit._quantity_type()
 
-    def __new__(cls, real_value, unit_str=None, quantity_map=None):
+    def __new__(cls, real_value, unit_str=None, quantity_map=None, dimensions=None):
         if unit_str and quantity_map:
             raise ValueError("Either unit or quantity_map is allowed.")
         if quantity_map:
             unit_str = get_unit_from_map(quantity_map)
+        if dimensions:
+            if len(dimensions) > 9:
+                raise ValueError("Number of dimensions should be <= 9")
+            dims_to_float = [float(dim) for dim in dimensions]
+            while len(dims_to_float) < 9:
+                dims_to_float.append(0.0)
+            unit_str = get_si_unit_from_dim(dim_list=dims_to_float)
```

Comment:
```suggestion
            unit_str = get_si_unit_from_dim(dim_list=dims_to_float + ((9-len(dims_to_float))*[0]))
```
Alternatively you could do all the padding inside `get_si_unit_from_dim`

Reply:
Done. Thank you.